epoxy-adapter/src/main/java/com/airbnb/epoxy/Carousel.java

```diff
@@ -318,37 +333,115 @@ public void setPadding(@Nullable Padding padding) {
+        @Dimension(unit = Dimension.DP) int itemSpacingDp) {
+      return new Padding(paddingDp, paddingDp, paddingDp, paddingDp, itemSpacingDp, PaddingType.DP);
+    }
 
     /**
-     * @param paddingPx     Padding in pixels to add on all sides of the carousel
+     * @param topDp Top padding in dp.
+     * @param bottomDp Bottom padding in dp.
+     * @param leftDp Left padding in dp.
+     * @param rightDp Right padding in dp.
+     * @param itemSpacingDp Space in dp to add between each carousel item. Will be implemented via
+     *     an item decoration.
+     */
+    public static Padding dp(
+        @Dimension(unit = Dimension.DP) int topDp,
+        @Dimension(unit = Dimension.DP) int bottomDp,
+        @Dimension(unit = Dimension.DP) int leftDp,
+        @Dimension(unit = Dimension.DP) int rightDp,
+        @Dimension(unit = Dimension.DP) int itemSpacingDp) {
+      return new Padding(topDp, bottomDp, leftDp, rightDp, itemSpacingDp, PaddingType.DP);
+    }
+
+    /**
+     * @param paddingPx Padding in pixels to add on all sides of the carousel
      * @param itemSpacingPx Space in pixels to add between each carousel item. Will be implemented
-     *                      via an item decoration.
+     *     via an item decoration.
      */
     public Padding(@Px int paddingPx, @Px int itemSpacingPx) {
-      this(paddingPx, paddingPx, paddingPx, paddingPx, itemSpacingPx);
+      this(paddingPx, paddingPx, paddingPx, paddingPx, itemSpacingPx, PaddingType.PX);
     }
 
     /**
-     * @param topPx         Top padding in pixels.
-     * @param bottomPx      Bottom padding in pixels.
-     * @param leftPx        Left padding in pixels.
-     * @param rightPx       Right padding in pixels.
+     * @param topPx Top padding in pixels.
+     * @param bottomPx Bottom padding in pixels.
+     * @param leftPx Left padding in pixels.
+     * @param rightPx Right padding in pixels.
      * @param itemSpacingPx Space in pixels to add between each carousel item. Will be implemented
-     *                      via an item decoration.
+     *     via an item decoration.
+     */
+    public Padding(
+        @Px int topPx, @Px int bottomPx, @Px int leftPx, @Px int rightPx, @Px int itemSpacingPx) {
+      this(topPx, bottomPx, leftPx, rightPx, itemSpacingPx, PaddingType.PX);
+    }
+
+    /**
+     * @param top Top padding.
+     * @param bottom Bottom padding.
+     * @param left Left padding.
+     * @param right Right padding.
+     * @param itemSpacing Space to add between each carousel item. Will be implemented via an item
+     *     decoration.
+     * @param paddingType Unit / Type of the given paddings/ itemspacing.
      */
-    public Padding(@Px int topPx, @Px int bottomPx, @Px int leftPx, @Px int rightPx,
-        @Px int itemSpacingPx) {
-
-      this.topPx = topPx;
-      this.bottomPx = bottomPx;
-      this.leftPx = leftPx;
-      this.rightPx = rightPx;
-      this.itemSpacingPx = itemSpacingPx;
+    public Padding(
```

Comment:
I don't think we need to expose this constructor since all types can be set directly already, and having this additional way to do it might be confusing